modules/402-ingress-nginx/templates/controller/controller.yaml

```diff
@@ -111,6 +102,11 @@ spec:
       maxUnavailable: 1
       maxSurge: 0
     {{- end }}
+  {{- if and (eq $crd.spec.inlet "HostWithFailover") (not $failover) }}
+  lifecycle:
```

Comment:
NIT: the link to a doc would be helpful
https://openkruise.io/docs/user-manuals/advanceddaemonset#lifecycle-hook

Reply:
done